### PR TITLE
New Feature: rstudio-server

### LINKED ISSUE #71: New feature: RStudio Server
There may be demand to use RStudio Server in combination with r-apt feature or r-rig feature.

### PR DESCRIPTION
**Pull request by eitsupi**
Close #71

### CODE REVIEW

**Comment by eitsupi** [comment]
@rokroskar Could you take a look at this?

**Comment by rokroskar** [comment]
Hi @eitsupi, this is looking great! I tested the build on my mac also with a non-privileged user. This is the `devcontainer.json` I used:

```json
{
	"image": "ubuntu",
	"features": {
		"./rstudio": {
			"version": "daily"
		},
		"ghcr.io/devcontainers/features/common-utils": {
			"username": "rstudio",
			"userGid": 1000,
			"userUid": 1000
		}
	},
	"remoteUser": "rstudio"
}
```

I have to make these modifications inside the container to run rstudio: 

```
$ mkdir /tmp/dbconf/db.conf
$ cat <<EOF>/tmp/dbconf/dbconf.conf
provider=sqlite
directory=/tmp/dbconf
EOF
$ mkdir /tmp/rstudio-data
```

Then I can run it with 

```
rserver --auth-none=0 \
--www-frame-origin=same \
--www-port=56559 \
--www-verify-user-agent=0 \
--secure-cookie-key-file=/tmp/key-file \
--server-user=rstudio \
--database-config-file=/tmp/dbconf/dbconf.conf \
--server-data-dir=/tmp/rstudio-data
```

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -119,6 +122,7 @@ jobs:
           - r-packages
           - r-rig
           - renv-cache
+          - rstudio-server
```
[the remark this answers is not in the document]
Comment (eitsupi) [reply .github/workflows/test.yaml]: Ok, it installs from the `rstudio/rstudio` repo, so `rstudio` makes sense.

[the remark this answers is not in the document]
Comment (eitsupi) [reply .github/workflows/test.yaml]: I again think that it is confusing with RStudio Desktop, so I still think it is better to clearly indicate "server".

**Comment by eitsupi** [comment]
It seems working fine with the start command `rserver` now.

image

image

built from

```json
{
	"image": "mcr.microsoft.com/devcontainers/base:ubuntu",
	"features": {
		"./rstudio-server": {},
		"ghcr.io/rocker-org/devcontainer-features/r-apt": {}
	}
}
```

**Comment by eitsupi** [comment]
TODO: https://github.com/rocker-org/devcontainer-images/blob/eb515c1e6ad3b751770a66207eba2a629177a8c1/src/rstudio/Dockerfile#L5

File: .github/workflows/test.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (rokroskar) [reply .github/workflows/test.yaml:125]: My thinking was that it would be consistent with the rocker docker image names but what you say also makes sense.

**Comment by eitsupi** [comment]
> TODO: [rocker-org/devcontainer-images@`eb515c1`/src/rstudio/Dockerfile#L5](https://github.com/rocker-org/devcontainer-images/blob/eb515c1e6ad3b751770a66207eba2a629177a8c1/src/rstudio/Dockerfile#L5)

This might make more sense to set up using a lifecycle script.

**Comment by eitsupi** [comment]
It seems to be working, so I will merge it.

**Comment by rokroskar** [comment]
Just tested it with my setup and it works great! Thanks for putting this together! 🙌

**Pull request merged**